src/v/redpanda/application.cc

```diff
@@ -1485,7 +1494,13 @@ void application::start_runtime_services(
     _co_group_manager.invoke_on_all(&kafka::group_manager::start).get();
 
     syschecks::systemd_message("Starting controller").get();
-    controller->start(cluster_discovery.initial_seed_brokers()).get0();
+    controller
+      ->start(
+        stored_cluster_uuid.has_value() ? std::vector<model::broker>{}
+                                        : cd.initial_seed_brokers(),
+        stored_cluster_uuid)
```

Comment:
Can we make this logic, deciding about the initial servers a part of `cluster_discovery` ?

Reply:
It was there originally, and it looked like this:
https://github.com/andrwng/redpanda/blob/cluster_bootstrap/src/v/cluster/cluster_discovery.cc#L85-L88
https://github.com/andrwng/redpanda/blob/cluster_bootstrap/src/v/redpanda/application.cc#L1517

That seemed a bit awkward to me to pass an argument to a function only to return empty if it's true. So I've decided to take it out. After all, it depends on the perspective to the `cluster_discovery::intial_seed_brokers`: is it about geting seed brokers for unstarted cluster only, or about doing that _and_ also about embedding the knowledge whether there is a cluster into its result? The latter seemed to me a bit like a function doing more than one thing.